sql_file3.sql

```diff
@@ -0,0 +1,705 @@
+(4, '2024-09-30 23:15:00');
+
+SELECT FlightID,DepartureTime AS DepartureTimeUTC,DepartureTime AT TIME ZONE 'Eastern Standard Time' AS DepartureTimeEST
+FROM Flights;
+
+
+-- 43 Add 3 months to the current date.
+
+SELECT DATEADD(MONTH,3,GETDATE())  AS ThirdMonthLater;
+
+-- 44 From an 'Employees' table, write a query to calculate each employee's retirement date (65 years from their 'DateOfBirth').
+
+CREATE TABLE Employees2 (
+    EmployeeID INT PRIMARY KEY,
+    EmployeeName VARCHAR(100),
+    DateOfBirth DATE  -- Column to store the date of birth
+);
+
+INSERT INTO Employees2 (EmployeeID, EmployeeName, DateOfBirth) VALUES
+(1, 'John Doe', '1980-01-15'),
+(2, 'Jane Smith', '1975-05-22'),
+(3, 'Alice Johnson', '1990-10-10'),
+(4, 'Bob Brown', '1960-02-28');
+
+SELECT EmployeeID , EmployeeName ,DateOfBirth , DATEADD(YEAR,65,DateOfBirth) AS RetirementDate
+FROM Employees2;
+
+
+-- 45 Calculate the number of days between '2023-01-01' and '2023-12-31
+
+SELECT DATEDIFF(DAY,'2023-01-01','2023-12-31') AS DateDifference;
+
+-- 46 From an 'Orders' table, find the average number of days between order date and ship date.
+
+CREATE TABLE Orders4 (
+    OrderID INT PRIMARY KEY,
+    OrderDate DATE,
+    ShipDate DATE
+);
+
+INSERT INTO Orders4 (OrderID, OrderDate, ShipDate) VALUES
+(1, '2024-09-01', '2024-09-05'),
+(2, '2024-09-02', '2024-09-10'),
+(3, '2024-09-03', '2024-09-04'),
+(4, '2024-09-04', '2024-09-08'),
+(5, '2024-09-05', '2024-09-12');
+
+SELECT DATEDIFF(DAY,OrderDate,ShipDate) AS DateDifference
+FROM Orders4;
+
+--47 Extract the month number from the date '2023-09-15'
+
+SELECT MONTH('2023-09-15') AS MonthOfDate
+
+-- 48 From a 'Sales' table, write a query to group total sales by the quarter of the sale date.
+
+CREATE TABLE Sales (
+    SaleID INT PRIMARY KEY,
+    SaleDate DATE,
+    TotalAmount DECIMAL(10, 2) 
+);
+
+INSERT INTO Sales (SaleID, SaleDate, TotalAmount) VALUES
+(1, '2023-01-15', 150.00),
+(2, '2023-04-22', 200.00),
+(3, '2023-07-10', 300.00),
+(4, '2023-10-05', 400.00),
+(5, '2023-01-20', 250.00),
+(6, '2023-04-25', 350.00);
+
+SELECT DATEPART(QUARTER,SaleDate) AS Quarters, SUM(TotalAmount) AS TotalOfEachQuarter
+FROM Sales
+GROUP BY DATEPART(QUARTER,SaleDate)
+ORDER BY DATEPART(QUARTER,SaleDate);
+
+
+
+-- 49 Extract the year from the current date.
+
+SELECT YEAR(GETDATE()) AS YearOfDate;
+
+-- 50 From an 'Employees' table, find all employees hired in the year 2022
+
+CREATE TABLE Employees3 (
+    EmployeeID INT PRIMARY KEY,
+    EmployeeName VARCHAR(100),
+    HireDate DATE
+);
+
+INSERT INTO Employees3 (EmployeeID, EmployeeName, HireDate) VALUES
+(1, 'John Doe', '2022-01-15'),
+(2, 'Jane Smith', '2022-06-22'),
+(3, 'Alice Johnson', '2021-09-10'),
+(4, 'Bob Brown', '2022-11-05'),
+(5, 'Charlie Black', '2020-12-30');
+
+SELECT * 
+FROM Employees3
+WHERE YEAR(HireDate) = 2022;
+
+-- 51 Check if '2023-02-30' is a valid date.
+
+
+IF TRY_CONVERT(DATE,'2023-02-30') IS NULL
+	PRINT 'INVALID DATE'
+ELSE
+	PRINT 'VAILD DATE'
```

Comment:
Spelling correction: "VAILD" to "VALID."

Reply:
IF TRY_CONVERT(DATE,'2023-02-30') IS NULL
	PRINT 'INVALID DATE'
ELSE
	PRINT 'VALID DATE'